hw/dv/dpi/usbdpi/usb_monitor.c

```diff
@@ -56,50 +60,112 @@ struct mon_ctx {
   unsigned char bytes[MON_BYTES_SIZE + 2];
 };
 
-void *monitor_usb_init(void) {
-  struct mon_ctx *mon = (struct mon_ctx *)calloc(1, sizeof(struct mon_ctx));
-  assert(mon);
+/**
+ * Create and initialise a USB monitor instance
+ */
+usb_monitor_ctx_t *usb_monitor_init(const char *filename) {
+  usb_monitor_ctx_t *mon =
+      (usb_monitor_ctx_t *)calloc(1, sizeof(usb_monitor_ctx_t));
+  USBDPI_ASSERT(mon);
 
   mon->state = MS_IDLE;
   mon->driver = M_NONE;
   mon->pu = 0;
   mon->line = 0;
-  return (void *)mon;
+
+  mon->file = fopen(filename, "w");
+  if (!mon->file) {
+    fprintf(stderr, "USBDPI: Unable to open monitor file at %s: %s\n", filename,
+            strerror(errno));
+    free(mon);
+    return NULL;
+  }
+
+  // more useful for tail -f
+  setlinebuf(mon->file);
+  printf(
+      "\nUSBDPI: Monitor output file created at %s. Works well with tail:\n"
+      "$ tail -f %s\n",
+      filename, filename);
+
+  return mon;
+}
+
+/**
+ * Finalise a USB monitor
```

Comment:
Nit: Finalize

Reply:
I've made a mental note to try to use US spellings in future, thanks